portfolio/src/main/webapp/script.js

```diff
@@ -26,19 +26,20 @@ function off() {
 async function getComments(value=2) {
   const response = await fetch('/data?comment-limit-choice=' + value);
   const content = await response.json();
+  console.log('getComments() content: ' + content.comments + ', ' + content.emails);
   
   const contentListElement = document.getElementById("comment-container");
   contentListElement.innerHTML = '';
-  for(let i = 0; i < content.length; i++) {
+  for(let i = 0; i < content.comments.length; i++) {
     contentListElement.appendChild(
-      createListElement(content[i]))
+      createListElement(content.emails[i], content.comments[i]));
```

Comment:
The current design assumes that comments.length == emails.length but you didn't verify that so you could be indexing outside of the emails array.

Instead of having two parallel arrays that you have to index into separately, it would be better to have a datastructure that contained an email and a comment together.  So you only index into it once and get all the content associated with that post.

The reason being is that if in the future you add a third element you don't need to update all places to use another third parallel array.  They get the "Post" object which has everything.

Reply:
Thanks, this is a good idea!